### PR TITLE
organisation changes

### PR DESCRIPTION
**Pull request by srikanthgudipudi**
Organisation work flow

### CODE REVIEW

File: src/app/modules/admin-core/services/organisation-service/organisation.service.ts [hunk 1]
```diff
@@ -32,12 +32,14 @@ export class OrganisationService {
...
-      data.status = 'Active';
-    }
-    return this.Http.get(environment.base_url + OrganisationConfig.blockOrganisation + orgid + '?status=' + data.status)
+  activate_deActivate_Organisation(data) {
+    return this.Http.post(environment.base_url + OrganisationConfig.blockOrganisation, data);
   }
+
+  // Update Organisation
+  updateOrganisation(data) {
+    console.log('uuuuuuuuuuuuu', data);
```
Comment (kiranharidas187) [inline src/app/modules/admin-core/services/organisation-service/organisation.service.ts]: remove logs

File: src/app/modules/organisations/organisation-edit-modal/organisation-edit-modal.component.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+</div>
+<mat-dialog-content class="mat-typography custom-dialog-content">
+  <div class="row">
+    <dynamic-form [fields]="formdata">
+    </dynamic-form>
+  </div>
+</mat-dialog-content>
+<mat-dialog-actions class="custom-dialog-actions" align="end">
+  <button mat-button mat-dialog-close>CANCEL</button>
+  <button mat-button style="background: #bababa;" (click)="onSubmit()" cdkFocusInitial>EDIT ORGANISATION</button>
```
Comment (kiranharidas187) [inline src/app/modules/organisations/organisation-edit-modal/organisation-edit-modal.component.html]: Update Organisation it should be

File: src/app/modules/organisations/organisation-edit-modal/organisation-edit-modal.component.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<div>
```
Comment (kiranharidas187) [inline src/app/modules/organisations/organisation-edit-modal/organisation-edit-modal.component.html]: cant we resue the Add organization modal??

File: src/app/modules/users/users-edit/users-edit.component.ts [hunk 4]
```diff
@@ -134,6 +157,23 @@ export class UsersEditComponent implements OnInit {
...
+  deleteUser(data) {
+    let removeUser = {
+      userId: this.userId,
+      organisationId: data.value
+    }
+    this.usersService.removeUserFromOrganisation(removeUser).subscribe(data => {
+      this.commonServiceService.commonSnackBar(data['message'], 'Dismiss', 'top', 10000)
+      this.getUserDetails();
+    }, error => {
+
```
Comment (kiranharidas187) [inline src/app/modules/users/users-edit/users-edit.component.ts]: Error not handled

File: src/app/modules/admin-core/services/organisation-service/organisation.service.ts (hunk 1 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-core/services/organisation-service/organisation.service.ts]: Removed

File: src/app/modules/organisations/organisation-edit-modal/organisation-edit-modal.component.html (hunk 2 above)
Comment (srikanthgudipudi) [reply src/app/modules/organisations/organisation-edit-modal/organisation-edit-modal.component.html]: changed

File: src/app/modules/organisations/organisation-edit-modal/organisation-edit-modal.component.html (hunk 3 above)
Comment (srikanthgudipudi) [reply src/app/modules/organisations/organisation-edit-modal/organisation-edit-modal.component.html]: tired but some errors are coming and also we need to keep so many conditions

File: src/app/modules/users/users-edit/users-edit.component.ts (hunk 4 above)
Comment (srikanthgudipudi) [reply src/app/modules/users/users-edit/users-edit.component.ts]: handled

File: src/app/modules/organisations/organisation-edit-modal/organisation-edit-modal.component.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import { MatDialog, MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { DynamicFormComponent } from '../../admin-shared';
+import { FieldConfig } from "../../admin-shared/field.interface";
+import { OrganisationService, CommonServiceService } from '../../admin-core';
+
+
+@Component({
+  selector: 'app-organisation-edit-modal',
+  templateUrl: './organisation-edit-modal.component.html',
+  styleUrls: ['./organisation-edit-modal.component.scss']
```
Comment (kiranharidas187) [inline src/app/modules/organisations/organisation-edit-modal/organisation-edit-modal.component.ts]: We discussed to same modal for edit and create, please complete it

File: src/app/modules/organisations/create-organisation/create-organisation.component.html [hunk 6]
```diff
@@ -1,7 +1,8 @@
 <div>
   <mat-toolbar role="toolbar" class="task-header">
     <mat-icon style="color: white;">account_balance</mat-icon>
-    <h5 class="title">Add Organisation</h5>
+    <h5 *ngIf="action == 'Add'" class="title">Add Organisation</h5>
+    <h5 *ngIf="action == 'Edit'" class="title">Update Organisation</h5>
```
Comment (kiranharidas187) [inline src/app/modules/organisations/create-organisation/create-organisation.component.html]: <h5  class="title">{{action == 'Add' ? Add Organisation : Update Organisation}} </h5>

Follow this pattern

File: src/app/modules/organisations/create-organisation/create-organisation.component.html [hunk 7]
```diff
@@ -12,5 +13,7 @@ <h5 class="title">Add Organisation</h5>
 </mat-dialog-content>
 <mat-dialog-actions class="custom-dialog-actions" align="end">
   <button mat-button mat-dialog-close>CANCEL</button>
-  <button mat-button style="background: #bababa;" (click)="onSubmit()" cdkFocusInitial>ADD ORGANISATION</button>
+  <button *ngIf="action == 'Add'" mat-button class="p-button" (click)="onSubmit()" cdkFocusInitial>ADD ORGANISATION</button>
+  <button *ngIf="action == 'Edit'" mat-button class="p-button" (click)="onEditSubmit()" cdkFocusInitial>UPDATE ORGANISATION</button>
```
Comment (kiranharidas187) [inline src/app/modules/organisations/create-organisation/create-organisation.component.html]: Follow the above-metioned pattern here also

File: src/app/modules/organisations/create-organisation/create-organisation.component.ts [hunk 8]
```diff
@@ -14,13 +14,18 @@ export class CreateOrganisationComponent implements OnInit {
   @ViewChild(DynamicFormComponent) form: DynamicFormComponent;
```
Comment (kiranharidas187) [inline src/app/modules/organisations/create-organisation/create-organisation.component.ts:14]: Rename the component name to editAndCreateOrganization

File: src/app/modules/organisations/organisation-edit/organisation-edit.component.ts [hunk 9]
```diff
@@ -53,8 +59,70 @@ export class OrganisationEditComponent implements OnInit {
...
+      if (this.confirmPopupResult) {
+        this.activate_deActivate_Organisation(this.editOrganisationDetails);
+      } else {
+        this.dialog.closeAll();
+      }
+    });
+  }
+
+  // activate and deActivate User
+  activate_deActivate_Organisation(data) {
```
Comment (kiranharidas187) [inline src/app/modules/organisations/organisation-edit/organisation-edit.component.ts]: Use Camel case

File: src/app/modules/organisations/organisation-edit/organisation-edit.component.ts [hunk 10]
```diff
@@ -53,8 +59,70 @@ export class OrganisationEditComponent implements OnInit {
...
+      , {
+        disableClose: true,
+        width: '50%',
+        data: { fieldsForOrganisation }
+      });
 
-      console.log('confirmPopupResult', this.confirmPopupResult);
+    dialogRef.afterClosed().subscribe(result => {
+      console.log('The dialog was closed');
+      this.getOrganisationDetails();
```
Comment (kiranharidas187) [inline src/app/modules/organisations/organisation-edit/organisation-edit.component.ts:125]: Don't fetch Organisation details if the user clicks on cancel of Edit modal

File: src/app/modules/organisations/organisations-list/organisations-list.component.ts [hunk 11]
```diff
@@ -155,14 +155,23 @@ export class OrganisationsListComponent implements OnInit {
 
   // Activate and Deactivate User
   activate_deActivate_Organisation(data) {
```
Comment (kiranharidas187) [inline src/app/modules/organisations/organisations-list/organisations-list.component.ts]: camelCase

File: src/app/modules/organisations/create-organisation/create-organisation.component.html (hunk 6 above)
Comment (srikanthgudipudi) [reply src/app/modules/organisations/create-organisation/create-organisation.component.html]: Changed

File: src/app/modules/organisations/create-organisation/create-organisation.component.html (hunk 7 above)
Comment (srikanthgudipudi) [reply src/app/modules/organisations/create-organisation/create-organisation.component.html]: Even methods are different , here we need two conditions

File: src/app/modules/organisations/create-organisation/create-organisation.component.ts (hunk 8 above)
Comment (srikanthgudipudi) [reply src/app/modules/organisations/create-organisation/create-organisation.component.ts:14]: changed

File: src/app/modules/organisations/organisation-edit/organisation-edit.component.ts (hunk 9 above)
Comment (srikanthgudipudi) [reply src/app/modules/organisations/organisation-edit/organisation-edit.component.ts]: changed

File: src/app/modules/organisations/organisation-edit/organisation-edit.component.ts (hunk 10 above)
Comment (srikanthgudipudi) [reply src/app/modules/organisations/organisation-edit/organisation-edit.component.ts:125]: removed

File: src/app/modules/organisations/organisations-list/organisations-list.component.ts (hunk 11 above)
Comment (srikanthgudipudi) [reply src/app/modules/organisations/organisations-list/organisations-list.component.ts]: Changed

File: src/app/modules/organisations/createandEdit-organisation/createandEdit-organisation.component.html [hunk 12]
```diff
@@ -12,5 +12,9 @@ <h5 class="title">Add Organisation</h5>
 </mat-dialog-content>
 <mat-dialog-actions class="custom-dialog-actions" align="end">
   <button mat-button mat-dialog-close>CANCEL</button>
-  <button mat-button style="background: #bababa;" (click)="onSubmit()" cdkFocusInitial>ADD ORGANISATION</button>
+  <button *ngIf="action == 'Add'" mat-button class="p-button" (click)="onSubmit()" cdkFocusInitial>ADD
+    ORGANISATION</button>
+  <button *ngIf="action == 'Edit'" mat-button class="p-button" (click)="onEditSubmit()" cdkFocusInitial>UPDATE
+    ORGANISATION</button>
```
Comment (kiranharidas187) [inline src/app/modules/organisations/createandEdit-organisation/createandEdit-organisation.component.html:18]: User ternary operator here also

File: src/styles.scss [hunk 13]
```diff
@@ -2638,6 +2638,23 @@ th.mat-sort-header-sorted {
 //   height: 550px;
 // }
 
+.example-container {
```
Comment (kiranharidas187) [inline src/styles.scss]: Use valid class name

**Pull request merged**